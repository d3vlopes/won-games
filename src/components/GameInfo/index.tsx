import { AddShoppingCart } from '@styled-icons/material/AddShoppingCart'
import { HeartOutline } from '@styled-icons/evaicons-outline/HeartOutline'

import formatPrice from 'utils/format-price'

import Heading from 'components/Heading'
import Ribbon from 'components/Ribbon'
import Button from 'components/Button'

import * as S from './styles'

export type GameInfoProps = {
  title: string
  description: string
  price: number
}

const GameInfo = ({ title, description, price }: GameInfoProps) => (
  <S.Wrapper>
    <Heading lineBottom color="black">
      {title}
    </Heading>

    <Ribbon color="secondary">{formatPrice(price)}</Ribbon>

    <S.Description>{description}</S.Description>

    <S.ButtonsWrapper>
      <Button size="large" icon={<AddShoppingCart />}>
        Adicionar ao carrinho
      </Button>

      <Button icon={<HeartOutline />} size="large" minimal>
        Favorito
      </Button>
    </S.ButtonsWrapper>
  </S.Wrapper>
)

export default GameInfo
