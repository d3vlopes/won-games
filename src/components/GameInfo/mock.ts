export default {
  id: '1',
  title: 'Borderlands 3',
  price: 215,
  description:
    'Experience the epic space strategy games tag redefined the RTS genre. Control you fleet and build an armada across more than 30 single-player missions.'
}
