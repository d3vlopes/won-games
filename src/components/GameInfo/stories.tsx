import { Story, Meta } from '@storybook/react/types-6-0'

import GameInfo, { GameInfoProps } from '.'
import mockGames from './mock'

export default {
  title: 'Game/GameInfo',
  component: GameInfo,
  parameters: {
    backgrounds: {
      default: 'won-dark'
    }
  },
  args: mockGames
} as Meta

export const Default: Story<GameInfoProps> = (args) => (
  <div style={{ maxWidth: '144rem', padding: '1.5rem' }}>
    <GameInfo {...args} />
  </div>
)
