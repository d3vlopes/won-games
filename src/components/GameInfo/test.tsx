import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import GameInfo from '.'

const props = {
  title: 'My Game Title',
  description: 'Game Description',
  price: '210,00'
}

describe('<GameInfo />', () => {
  it('should render game informations', () => {
    const { container } = renderWithTheme(<GameInfo {...props} />)

    // Verifica o heading(title)
    expect(
      screen.getByRole('heading', { name: props.title })
    ).toBeInTheDocument()

    // Verifica a description
    expect(screen.getByText(props.description)).toBeInTheDocument()

    // Verifica o price
    expect(screen.getByText(/r\$210,00/i)).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render buttons', () => {
    renderWithTheme(<GameInfo {...props} />)

    // Verifica o button add to cart
    expect(
      screen.getByRole('button', { name: /adicionar ao carrinho/i })
    ).toBeInTheDocument()

    // Verifica o button wishlist
    expect(
      screen.getByRole('button', { name: /favoritos/i })
    ).toBeInTheDocument()
  })
})
