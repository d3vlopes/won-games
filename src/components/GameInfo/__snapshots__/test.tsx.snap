// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GameInfo /> should render game informations 1`] = `
.c7 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c1 {
  color: #030517;
  position: relative;
  margin-bottom: 3.2rem;
  font-size: 2.0rem;
}

.c1::after {
  position: absolute;
  left: 0;
  bottom: -1rem;
  content: '';
  width: 5rem;
  border-bottom: 0.5rem solid #F231A5;
}

.c3 {
  position: absolute;
  background-color: #3CD3C1;
  top: 1.6rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-weight: 600;
  color: #FAFAFA;
  font-size: 1.4rem;
  padding: 0 2.4rem;
  height: 3.6rem;
  right: -2rem;
  background-color: #3CD3C1;
}

.c3::before {
  border-left-color: #1f8a7d;
  border-top-color: #1f8a7d;
}

.c3::before {
  content: '';
  position: absolute;
  right: 0;
  border-style: solid;
  border-left-width: 0rem;
  border-right-color: transparent;
  border-bottom-color: transparent;
  border-bottom-width: 1rem;
}

.c3::before {
  top: 3.6rem;
  border-top-width: 1rem;
  border-right-width: 2rem;
}

.c3::before {
  border-left-color: #1f8a7d;
  border-top-color: #1f8a7d;
}

.c6 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background: linear-gradient(180deg,#ff5f5f 0%,#f062c0 50%);
  color: #FAFAFA;
  border: 0;
  cursor: pointer;
  border-radius: 0.4rem;
  padding: 0.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  height: 5rem;
  font-size: 1.6rem;
  padding: 0.8rem 4.8rem;
}

.c6:hover {
  background: linear-gradient(180deg,#e35565 0%,#d958a6 50%);
}

.c6 svg {
  width: 1.5rem;
}

.c6 svg + span {
  margin-left: 0.8rem;
}

.c8 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background: linear-gradient(180deg,#ff5f5f 0%,#f062c0 50%);
  color: #FAFAFA;
  border: 0;
  cursor: pointer;
  border-radius: 0.4rem;
  padding: 0.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  height: 5rem;
  font-size: 1.6rem;
  padding: 0.8rem 4.8rem;
  background: none;
  color: #F231A5;
}

.c8:hover {
  background: none;
}

.c8 svg {
  width: 1.5rem;
}

.c8 svg + span {
  margin-left: 0.8rem;
}

.c8:hover {
  color: #e20e8d;
}

.c0 {
  position: relative;
  background: #FAFAFA;
  padding: 2.4rem;
}

.c0 .c2 {
  right: -1rem;
}

.c0 .c2:before {
  border-right-width: 1rem;
}

.c4 {
  font-size: 1.4rem;
  color: #8F8F8F;
  margin-bottom: 2.4rem;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c5 > button {
  width: 100%;
  margin-bottom: 0.8rem;
}

@media (min-width:768px) {
  .c1 {
    font-size: 2.8rem;
  }
}

@media (min-width:768px) {
  .c0 .c2 {
    right: 2.4rem;
    top: 2.4rem;
    font-size: 1.8rem;
  }

  .c0 .c2:before {
    border: none;
  }
}

@media (min-width:768px) {
  .c4 {
    max-width: 77rem;
  }
}

@media (min-width:768px) {
  .c5 {
    -webkit-flex-direction: row-reverse;
    -ms-flex-direction: row-reverse;
    flex-direction: row-reverse;
  }

  .c5 > button {
    width: initial;
    margin-bottom: 0;
  }
}

<div
  class="c0"
>
  <h2
    class="c1"
    color="black"
  >
    My Game Title
  </h2>
  <div
    class="c2 c3"
    color="secondary"
  >
    R$210,00
  </div>
  <p
    class="c4"
  >
    Game Description
  </p>
  <div
    class="c5"
  >
    <button
      class="c6"
    >
      <svg
        aria-hidden="true"
        class="c7"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0zm18.31 6l-2.76 5z"
          fill="none"
        />
        <path
          d="M11 9h2V6h3V4h-3V1h-2v3H8v2h3v3zm-4 9c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zm10 0c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2zm-9.83-3.25l.03-.12.9-1.63h7.45c.75 0 1.41-.41 1.75-1.03l3.86-7.01L19.42 4h-.01l-1.1 2-2.76 5H8.53l-.13-.27L6.16 6l-.95-2-.94-2H1v2h2l3.6 7.59-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h12v-2H7.42c-.13 0-.25-.11-.25-.25z"
        />
      </svg>
      <span>
        Adicionar ao carrinho
      </span>
    </button>
    <button
      class="c8"
    >
      <svg
        aria-hidden="true"
        class="c7"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          data-name="Layer 2"
        >
          <path
            d="M12 21a1 1 0 01-.71-.29l-7.77-7.78a5.26 5.26 0 010-7.4 5.24 5.24 0 017.4 0L12 6.61l1.08-1.08a5.24 5.24 0 017.4 0 5.26 5.26 0 010 7.4l-7.77 7.78A1 1 0 0112 21zM7.22 6a3.2 3.2 0 00-2.28.94 3.24 3.24 0 000 4.57L12 18.58l7.06-7.07a3.24 3.24 0 000-4.57 3.32 3.32 0 00-4.56 0l-1.79 1.8a1 1 0 01-1.42 0L9.5 6.94A3.2 3.2 0 007.22 6z"
            data-name="heart"
          />
        </g>
      </svg>
      <span>
        Favoritos
      </span>
    </button>
  </div>
</div>
`;
